feel free to fix it better way.
like checking for $top_srcdir/config/rules.mk in every loop.

--- mozilla-1.8/extensions/ipc/makemake~	2005-04-02 16:32:47.000000000 +0300
+++ mozilla-1.8/extensions/ipc/makemake	2005-04-02 16:32:49.000000000 +0300
@@ -17,7 +17,7 @@
    echo $wd
 
    set top_srcdir = ""
-   while (("$wd" != "") && (${wd:t} != "mozilla"))
+   while (("$wd" != "") && (${wd:t} != "mozilla") && (${wd:t} != "mozilla-1.8"))
       if ("$top_srcdir" == "") then
          set top_srcdir = ".."
       else
--- mozilla-1.8/extensions/enigmail/makemake~	2005-04-02 16:32:47.000000000 +0300
+++ mozilla-1.8/extensions/enigmail/makemake	2005-04-02 16:32:49.000000000 +0300
@@ -17,7 +17,7 @@
    echo $wd
 
    set top_srcdir = ""
-   while (("$wd" != "") && (${wd:t} != "mozilla"))
+   while (("$wd" != "") && (${wd:t} != "mozilla") && (${wd:t} != "mozilla-1.8"))
       if ("$top_srcdir" == "") then
          set top_srcdir = ".."
       else
